lib/Driver/ToolChains.cpp

```diff
@@ -1319,6 +1319,184 @@ toolchains::GenericUnix::constructInvocation(const LinkJobAction &job,
+    case llvm::Triple::thumb:
+    case llvm::Triple::thumbeb:
+    // BFD linker has issues wrt relocation of the protocol conformance
+    // section on these targets, it also generates COPY relocations for
+    // final executables, as such, unless specified, we default to gold
+    // linker.
+      Linker = "gold";
+    }
+  }
+
+  if (!Linker.empty()) {
+    Arguments.push_back(context.Args.MakeArgString("-fuse-ld=" + Linker));
+  }
+
+  // Add the runtime library link path, which is platform-specific and found
+  // relative to the compiler.
+  // FIXME: Duplicated from CompilerInvocation, but in theory the runtime
+  // library link path and the standard library module import path don't
+  // need to be the same.
+  llvm::SmallString<128> RuntimeLibPath;
+
+  if (const Arg *A = context.Args.getLastArg(options::OPT_resource_dir)) {
+    RuntimeLibPath = A->getValue();
+  } else {
+    RuntimeLibPath = D.getSwiftProgramPath();
+    llvm::sys::path::remove_filename(RuntimeLibPath); // remove /swift
+    llvm::sys::path::remove_filename(RuntimeLibPath); // remove /bin
+    llvm::sys::path::append(RuntimeLibPath, "lib", "swift");
+  }
+  llvm::sys::path::append(RuntimeLibPath,
+                          getPlatformNameForTriple(getTriple()));
+
+ // On Linux and FreeBSD (really, ELF binaries) we need to add objects
+ // to provide markers and size for the metadata sections.
+  Arguments.push_back(context.Args.MakeArgString(
+    Twine(RuntimeLibPath) + "/" + getSectionMagicArch(getTriple()) + "/swift_begin.o"));
+  addPrimaryInputsOfType(Arguments, context.Inputs, types::TY_Object);
+  addInputsOfType(Arguments, context.InputActions, types::TY_Object);
+
+  context.Args.AddAllArgs(Arguments, options::OPT_Xlinker);
+  context.Args.AddAllArgs(Arguments, options::OPT_linker_option_Group);
+  context.Args.AddAllArgs(Arguments, options::OPT_F);
+
+  if (!context.OI.SDKPath.empty()) {
+    Arguments.push_back("--sysroot");
+    Arguments.push_back(context.Args.MakeArgString(context.OI.SDKPath));
+  }
+
+  Arguments.push_back("-L");
+  Arguments.push_back(context.Args.MakeArgString(RuntimeLibPath));
+
+  if (context.Args.hasArg(options::OPT_profile_generate)) {
+    SmallString<128> LibProfile(RuntimeLibPath);
+    llvm::sys::path::remove_filename(LibProfile); // remove platform name
+    llvm::sys::path::append(LibProfile, "clang", "lib");
+
+    llvm::sys::path::append(LibProfile, getTriple().getOSName(),
+                            Twine("libclang_rt.profile-") +
+                              getTriple().getArchName() +
+                              ".a");
+    Arguments.push_back(context.Args.MakeArgString(LibProfile));
+  }
+
+
+  // Explicitly set the linker target to "androideabi", as opposed to the
+  // llvm::Triple representation of "armv7-none-linux-android".
+  // This is the only ABI we currently support for Android.
+  Arguments.push_back("-target");
+  Arguments.push_back("armv7-none-linux-androideabi");
```

Comment:
Sorry, I thought I had left a reply for this, but it seems like it had slipped somehow. I took this target from the Android guides; see the section "Invoking the Compiler" [here](http://developer.android.com/ndk/guides/standalone_toolchain.html). There's a table on that page of architectures and corresponding values for `-target`:

| Architecture | Value |
| --- | --- |
| armeabi | -target armv5te-none-linux-androideabi |
| armeabi-v7a | -target armv7-none-linux-androideabi |
| arm64-v8a | -target aarch64-none-linux-android |
| x86	-target | i686-none-linux-android |
| x86_64 | -target x86_64-none-linux-android |
| mips | -target mipsel-none-linux-android |